ext/xml/xml.c

```diff
@@ -1406,6 +1415,7 @@ PHP_FUNCTION(xml_parser_set_option)
 			if (parser->toffset < 0) {
 				php_error_docref(NULL, E_WARNING, "tagstart ignored, because it is out of range");
 				parser->toffset = 0;
+				/* TODO Should this return false? */
```

Comment:
I think this should throw a `ValueError`, but it might not be the best idea to change that now for BC reasons.

Reply:
It probably should in the long run, but currently it's an error condition which returns `true` which is somewhat awkward.

Reply:
Right! Given that we currently type as `bool`, returning `false` in this case appears to be the best solution for now.

Reply:
Personally, I'd probably stay with true + add this to the PHP 9.0 warning promotion RFC when the time comes, but I don't mind returning `false`.

Reply:
That's also fair, as it has been this behaviour for a while... PHP does have some oddities.